src/magic_folder/uploader.py

```diff
@@ -271,16 +286,21 @@ def upload_local_snapshots(self):
         # the new snapshot gets lost. Perhaps this can be solved by storing each
         # LocalSnapshot in its own row than storing everything in a blob?
         # https://github.com/LeastAuthority/magic-folder/issues/197
+
+        # XXX above comment obsolete, right? (197 is fixed / database is normalized)
```

Comment:
The comment is not obsolete. `MagicFolderConfig.get_local_snapshot` actually reads all snapshots for that path, and `MagicFolderConfig.delete_localsnapshot` deletes all snapshots for that path. So:

1. read all snapshots
2. write all snapshots that existed at point (1)
3. delete all snapshots that exist at point (3)

If snapshots are created between 1 and 3, they'll get lost. There are a couple of ways to address this:
- change `delete_localsnapshot` to only delete the snapshots passed to it (incidentally, that functions needs to be changed to delete data from related tables, probably)
- change `write_snapshot_to_tahoe` to not be recursive, and upload them one at a time, deleting from the database as we go.
I prefer the second option, even though it may be more work, as it lets us make progress, even if we can't upload all the snapshots at once.

Reply:
https://github.com/LeastAuthority/magic-folder/issues/383 for the cleanup in related tables issue

Reply:
I changed the comment; just the part about "individual rows" is obsolete.

Most of the above should go into a ticket (it's related to the uploader not downloader anyway) so I clicked the "reference in new issue" button and it created #385